test/infrastructure/docker/internal/docker/machine.go

```diff
@@ -392,20 +395,15 @@ func (m *Machine) SetNodeProviderID(ctx context.Context) error {
 		return errors.Wrapf(ContainerNotRunningError{Name: kubectlNode.Name}, "unable to set NodeProviderID")
 	}
 
+	node := &corev1.Node{}
+	if err = c.Get(ctx, apimachinerytypes.NamespacedName{Name: m.ContainerName()}, node); err != nil {
+		return errors.Wrap(err, "failed to retrieve node")
+	}
+
 	log.Info("Setting Kubernetes node providerID")
-	patch := fmt.Sprintf(`{"spec": {"providerID": %q}}`, m.ProviderID())
-	cmd := kubectlNode.Commander.Command(
-		"kubectl",
-		"--kubeconfig", "/etc/kubernetes/kubelet.conf",
-		"patch",
-		"node", m.ContainerName(),
-		"--patch", patch,
-	)
-	lines, err := cmd.RunLoggingOutputOnFail(ctx)
-	if err != nil {
-		for _, line := range lines {
-			log.Info(line)
-		}
+
+	patch := client.RawPatch(apimachinerytypes.MergePatchType, []byte(fmt.Sprintf(`{"spec": {"providerID": %q}}`, m.ProviderID())))
+	if err = c.Patch(ctx, node, patch); err != nil {
 		return errors.Wrap(err, "failed update providerID")
 	}
```

Comment:
```suggestion
	patchHelper, err := patch.NewHelper(node, c)
	if err != nil {
		return err
	}
	
	node.Spec.ProviderID = m.ProviderID()

	if err = patchHelper.Patch(ctx, node); err != nil {
		return errors.Wrap(err, "failed update providerID")
	}
```

Let's try to use the patchHelper